src/unsafe_helpers.rs

```diff
@@ -37,3 +37,9 @@ pub(crate) fn mut_ref_to_ptr<T>(reference: &mut T) -> *mut () {
     let untyped_ptr: *mut () = ptr.cast();
     untyped_ptr
 }
+
+pub fn ref_as_ptr<T>(reference: &mut T) -> *mut () {
```

Comment:
Should this be `pub(crate)`?
How is this different from `mut_ref_to_ptr()`? They appear to have the same signature and implementation.

Reply:
This was also unused, thanks. Removed.

Years ago when I started this crate I added an `#![allow(dead_code)]` at the crate root because the lint was overly conservative and would complain about unused-but-public methods. I'm sure that's been fixed by now. I'll do another PR after this to re-enable it so we can prune stuff like this.